### ISSUE TITLE
Indexing doesn't work for medium sized projects on Windows 10 (files don't show up in 'Search in file' and 'Go to file')

### ISSUE THREAD

**Issue by ElranKadosh** [Microsoft/vscode#40076]
- VSCode Version: 1.18.1
- OS Version: Windows 10 Home Edition (64 bit)

Hi guys,

I have a **problem with indexing** for both my web(default support) and Python(with the official Python extension installed) projects (**22k files & 30k files projects**). For some of those(web projects) VSCode works fine but for the others I **can't find any files in Go to file and Search in files, excluding those I already opened** with the editor.

I **deleted any cache/settings** both for my **user and folder** and even **uninstalled**(then re-installed) VSCode but it still occurs in the same projects. I **tried the Insiders build(1.19.0-insider)** and it doesn't seem to be solved(yet) there. In addition, I **double verified** my user and folder **"search.exclude"** and **"files.exclude"** settings.

**NOTE: I originally commented here(requested to open a new issue):**

https://github.com/Microsoft/vscode/issues/39440#issuecomment-350900107

**It seems to be the exact _same problem_, but for medium sized projects**

Steps to Reproduce:

1. Open the problematic project

Reproduces without extensions: Yes

Thanks in advance,
Elran

**Comment by bsvipas** [comment]
Hi, try to add this to your settings: ```"files.useExperimentalFileWatcher": true```

Try to exclude as much as possible stuff like `tmp, temp, dist, build, node_modules, bower_components`, etc. all stuff which is not needed for you in search/go to file, they still will be available in file explorer.

**Comment by ElranKadosh** [comment]
Hi, thanks very much for the help. It works now! @bsvipas 

What made it work was to set the property "search.useRipgrep" to false. Or more accurately, what made the bug was the use of Ripgrep on Windows 10.

So the issue can be closed now :)

If you have the time for an optional question (I always have to know the whys and hows):
Except for the expected impact on performance, is there any other implication of setting this property to false? because I can't find any documentation on that.

**Comment by roblourens** [comment]
Thanks for trying that, I'd rather get to the bottom of why it doesn't work with ripgrep, if you don't mind a little more trouble shooting.

- In a folder where it doesn't work, can you show the output of the search output pane?

![image](https://user-images.githubusercontent.com/323878/32013959-89edf3e4-b971-11e7-8f3d-9f4b4b7790e9.png)

- What's different between the folders where it works, and where it doesn't?

**Comment by ElranKadosh** [comment]
![image](https://user-images.githubusercontent.com/18571677/33913676-7e517044-dfa3-11e7-85f4-f97a39b32c7d.png)

So to the point. That's the output for the folders where it doesn't work (and ripgrep is enabled, by default). There is no output when/for:
* a project where the indexing worked initially
* ripgrep is disabled

Because of that 'access denied' message, I thought it might worth the effort to try and run VSCode as an administrator, just in case.. but it didn't change anything.

Hope it helps! @roblourens

**Comment by roblourens** [comment]
Is that folder tracked in OneDrive?

**Comment by ElranKadosh** [comment]
Nope. I'd used to sync my workspace with OneDrive a few months ago, but I already learned this lesson ;)

**Comment by roblourens** [comment]
Hm... so what's different about that location? Is it all folders under Workspace that don't work?

**Comment by ElranKadosh** [comment]
Nope, some of them work and others don't. Though I only checked this for 4 folder (didn't work for 2 till I disabled 'ripgrep').

**Comment by roblourens** [comment]
I wonder if you copied that folder out of OneDrive earlier and it brought some weird attributes along with it?

**Comment by ElranKadosh** [comment]
I copied the problematic folders our of OneDrive. Regards your question.. I didn't notice anything on git, so I don't think so, but the only think certain in this work is uncertainty :)

If you heard about similar cases with folders that carry some evil out of OneDrive I'll be happy to know what should I look for(?)
@roblourens

**Comment by roblourens** [comment]
Well there is a bug around searching in some OneDrive folders - https://github.com/Microsoft/vscode/issues/35433, and it's suspicious that you get exactly the same symptoms and error message - `(os error 5)`.

Is it possible to try re-cloning the project into a different location? OneDrive can use special filesystem nodes which don't really show up anywhere else.

**Comment by PixelT** [comment]
I have the same issues as author have in main topic, **but** on MacOS
I have project with magento2 + couple of themes and the quick open option doesn't work for me (https://github.com/Microsoft/vscode/issues/40272)

**Comment by Srekel** [comment]
Same issue here, disabling ripgrep seems to work. 

- VSCode Version: Code 1.19.0 (816be6780ca8bd0ab80314e11478c48c70d09383, 2017-12-14T12:06:43.492Z)
- OS Version: Windows_NT x64 10.0.16299

**Comment by PixelT** [comment]
Unfortunatelly it stop works for me even if I have `"search.useRipgrep": false` ....

**Comment by roblourens** [comment]
@PixelT I asked you a question about your problem in #40272 earlier

**Comment by ElranKadosh** [comment]
Hi, sorry for the late response.

So.. I cloned the project folder(s) again from my remote repo and it seems to be working now.
I guess you were right, @roblourens , "OneDrive can use special filesystem nodes which don't really show up anywhere else.". Yes it does and this is why(I guess) it didn't work for me even though I moved our those folders from my One-Drive and why I didn't notice any change in my local git repo.

Thanks for the kind help, hope it will help others.
